ballerina/client.bal

```diff
@@ -56,320 +56,942 @@ public isolated client class Client {
-        http:Request request = check self.generateRequest(self.createPayload(parameters));
-        http:Response|error httpResponse = self.amazonSNSClient->post("/", request);
-        xml response = check handleResponse(httpResponse);
-        return xmlToHttpResponse(response);
+        http:Request request = check self.generateRequest(parameters);
+        _ = check sendRequest(self.amazonSNSClient, request);
     }
 
-    # Get values of SMS attributes.
+    # Publishes a message to an SNS topic, a phone number, or a mobile platform endpoint.
+    # 
+    # + target - The target (topic ARN, target ARN or phone number) to which to publish the message to
+    # + message - The message to publish. If you are publishing to a topic and you want to send the same message to all
+    #             transport protocols, include the text of the message as a `string` value. If you want to send 
+    #             different messages for each transport protocol, use a `MessageRecord` value
+    # + targetType - The type of target (topic, phone number, or application endpoint) to publish the message to
+    # + attributes - Attributes of the message
+    # + deduplicationId - Every message must have a unique `deduplicationId`, which is a token used for deduplication 
+    #                     of sent messages. If a message with a particular `deduplicationId` is sent successfully, any 
+    #                     message sent with the same `deduplicationId` during the 5-minute deduplication interval is 
+    #                     treated as a duplicate. If the topic has `contentBasedDeduplication` set, the system 
+    #                     generates a `deduplicationId` based on the contents of the message. Your `deduplicationId`
+    #                     overrides the generated one. Applies to FIFO topics only
+    # + groupId - Specifies the message group to which a message belongs to. Messages that belong to the same message 
+    #             group are processed in a FIFO manner (however, messages in different message groups might be processed
+    #             out of order). Every message must include a `groupId`. Applies to FIFO topics only
+    # + return - `PublishMessageResponse` or `sns:Error` in case of failure
+    isolated remote function publish(string target, Message message, TargetType targetType = TOPIC, 
+    map<MessageAttributeValue>? attributes = (), string? deduplicationId = (), string? groupId = ())
+        returns PublishMessageResponse|Error {
+        _ = check validatePublishParameters(target, targetType, groupId);
+        map<string> parameters = initiateRequest("Publish");
+
+        if (targetType == TOPIC) {
+            parameters["TopicArn"] = target;
+        } else if (targetType == ARN) {
+            parameters["TargetArn"] = target;
+        } else {
+            parameters["PhoneNumber"] = target;
+        }
+        
+        if message is string {
+            parameters["Message"] = <string>message;
```

Comment:
```suggestion
            parameters["Message"] = message;
```

Reply:
Fixed in [020b0a4](https://github.com/ballerina-platform/module-ballerinax-aws.sns/pull/40/commits/020b0a40d31509c3e7b499cf2ea5c620bbfc9b2b)